src/plugins/discover/public/application/apps/main/services/use_saved_search.ts

```diff
@@ -0,0 +1,328 @@
+              reset ? {} : refs.current.fieldCounts,
+              documents,
+              indexPattern
+            ),
+            hits: res.rawResponse.hits.total as number,
+          };
+
+          if (chartAggConfigs) {
+            const bucketAggConfig = chartAggConfigs!.aggs[1];
+            const tabifiedData = tabifyAggResponse(chartAggConfigs, res.rawResponse);
+            const dimensions = getDimensions(chartAggConfigs, data);
+            if (dimensions) {
+              if (bucketAggConfig && search.aggs.isDateHistogramBucketAggConfig(bucketAggConfig)) {
+                message.bucketInterval = bucketAggConfig.buckets?.getInterval();
+              }
+              message.chartData = buildPointSeriesData(tabifiedData, dimensions);
+            }
+          }
+          refs.current.fieldCounts = message.fieldCounts!;
+          refs.current.fetchStatus = message.state;
+          savedSearch$.next(message);
+        },
+        (error) => {
+          if (error instanceof Error && error.name === 'AbortError') return;
+          data.search.showError(error);
+          refs.current.fetchStatus = fetchStatuses.ERROR;
+          savedSearch$.next({
+            state: fetchStatuses.ERROR,
+            inspectorAdapters,
+            fetchError: error,
+          });
+        },
+        () => {
+          refs.current.autoRefreshDoneCb?.();
+          refs.current.autoRefreshDoneCb = undefined;
+        }
+      );
+    },
+    [
+      timefilter,
+      services,
+      stateContainer.appStateContainer,
+      searchSource,
+      indexPattern,
+      useNewFieldsApi,
+      state.hideChart,
+      state.interval,
+      data,
+      searchSessionManager,
+      savedSearch$,
+    ]
+  );
+
+  /**
+   * This part takes care of triggering the data fetching by creating and subscribing
+   * to an observable of various possible changes in state
+   */
+  useEffect(() => {
+    const fetch$ = merge(
+      refetch$,
+      filterManager.getFetches$(),
+      timefilter.getFetch$(),
+      timefilter.getAutoRefreshFetch$().pipe(
+        tap((done) => {
+          refs.current.autoRefreshDoneCb = done;
+        }),
+        filter(() => refs.current.fetchStatus !== fetchStatuses.LOADING)
+      ),
+      data.query.queryString.getUpdates$(),
+      searchSessionManager.newSearchSessionIdFromURL$
+    ).pipe(debounceTime(100));
+
+    const subscription = fetch$.subscribe({
+      next: (val) => {
+        fetchAll(val === 'reset');
+      },
+    });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [
+    data.query.queryString,
+    filterManager,
+    refetch$,
+    searchSessionManager.newSearchSessionIdFromURL$,
+    timefilter,
+    fetchAll,
+  ]);
+
+  /**
+   * Track state changes that should trigger a fetch
+   */
+  useEffect(() => {
+    let triggerFetch = false;
+    let resetFieldCounts = false;
+    if (state.hideChart !== refs.current.appState.hideChart && !state.hideChart) {
```

Comment:
This is very nitpicky, but we could improve readability by extracting these conditions in separate consts and then use the const names in one big if-statement. Eg:
```
const chartDisplayChanged = state.hideChart !== refs.current.appState.hideChart && !state.hideChart;
const chartIntervalChanged = state.interval !== refs.current.appState.interval;
const docTableSortChanged = isEqual(state.sort, refs.current.appState.sort)

if (chartDisplayChanged || chartIntervalChanged || docTableSortChanged) {
  triggerFetch = true;
}
```

Reply:
@majagrubic good point, always like readability improvements , how about this

```TypeScript
 const prevAppState = refs.current.appState;

  // chart was hidden, now it should be displayed, so data is needed
  const chartDisplayChanged = state.hideChart !== prevAppState.hideChart && !state.hideChart;
  const chartIntervalChanged = state.interval !== prevAppState.interval;
  const docTableSortChanged = !isEqual(state.sort, prevAppState.sort);
  const indexPatternChanged = !isEqual(state.index, prevAppState.index);

  refs.current.appState = state;
  if (chartDisplayChanged || chartIntervalChanged || docTableSortChanged || indexPatternChanged) {
    refetch$.next(indexPatternChanged ? 'reset' : '');
  }
```

Reply:
done that 👍